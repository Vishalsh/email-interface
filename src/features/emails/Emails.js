import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'

import { hasEmails } from 'utilities/emails';
import EmailListItem from 'features/emails/emailListItem/EmailListItem';
import emailsActions from './emails.actions';
import classes from './Emails.module.scss';

export const Emails = (props) => {
  const {
    mailbox,
    emails,
    updateEmailStatus,
    toggleEmailSelection
  } = props;

  return (
    <ul className={classes.emailList}>
      {
        emails.map((email) => (
          <EmailListItem key={email.id}
                         mailbox={mailbox}
                         email={email}
                         onClickEmail={updateEmailStatus}
                         onChangeCheckbox={toggleEmailSelection}/>
        ))
      }
    </ul>
  );
};

Emails.propTypes = {
  mailbox: PropTypes.string.isRequired,
  emails: PropTypes.array.isRequired,
  updateEmailStatus: PropTypes.func.isRequired
};

const mapStateToProps = (store, ownProps) => ({
  emails: hasEmails(store.emails) ? ownProps.ids.map(id => store.emails[id]) : []
});

const mapDispatchToProps = dispatch => bindActionCreators({
  updateEmailStatus: emailsActions.updateEmailStatus,
  toggleEmailSelection: emailsActions.toggleEmailSelection
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Emails);
