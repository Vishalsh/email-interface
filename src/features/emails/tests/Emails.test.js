import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Emails } from '../Emails';

describe('Emails', () => {
  const props = {
    emails: [{
      id: 1,
      subject: 'subject 1',
      from: 'MS Dhoni'
    }, {
      id: 1,
      subject: 'subject 2',
      from: 'Virat Kohli'
    }],
    mailbox: 'inbox'
  };
  const component = shallow(<Emails {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});