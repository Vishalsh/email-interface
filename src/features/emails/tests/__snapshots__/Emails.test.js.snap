// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emails should render the component 1`] = `
<ul
  className="emailList"
>
  <EmailListItem
    email={
      Object {
        "from": "MS Dhoni",
        "id": 1,
        "subject": "subject 1",
      }
    }
    key="1"
    mailbox="inbox"
    onChangeCheckbox={[MockFunction]}
    onClickEmail={[MockFunction]}
  />
  <EmailListItem
    email={
      Object {
        "from": "Virat Kohli",
        "id": 1,
        "subject": "subject 2",
      }
    }
    key="1"
    mailbox="inbox"
    onChangeCheckbox={[MockFunction]}
    onClickEmail={[MockFunction]}
  />
</ul>
`;
